packages/headless-components/services/src/react/core/ServiceList.tsx

```diff
@@ -0,0 +1,311 @@
+}
+
+/**
+ * Render props for Error component
+ */
+export interface ErrorRenderProps {
+  /** Error message */
+  error: string | null;
+}
+
+/**
+ * Component that renders content when there's an error loading services.
+ * Only displays its children when an error has occurred.
+ *
+ * @component
+ * @example
+ * ```tsx
+ * import { Services } from '@wix/services/components';
+ *
+ * function ServicesError() {
+ *   return (
+ *     <Services.Error>
+ *       {({ error }) => (
+ *         <div className="error-state">
+ *           <h3>Error loading services</h3>
+ *           <p>{error}</p>
+ *           <button onClick={() => window.location.reload()}>
+ *             Try Again
+ *           </button>
+ *         </div>
+ *       )}
+ *     </Services.Error>
+ *   );
+ * }
+ * ```
+ */
+export function Error(props: ErrorProps): React.ReactNode {
+  const { error } = useService(ServicesListServiceDefinition);
+  const errorValue = error.get();
+
+  if (errorValue) {
+    return typeof props.children === 'function'
+      ? props.children({ error: errorValue })
+      : props.children;
+  }
+
+  return null;
+}
+
+/**
+ * Props for ItemContent headless component
+ */
+export interface ItemContentProps {
+  /** Content to display for each service (can be a render function receiving service data or ReactNode) */
+  children:
+    | ((props: ItemContentRenderProps) => React.ReactNode)
+    | React.ReactNode;
+}
+
+/**
+ * Render props for ItemContent component
+ */
+export interface ItemContentRenderProps {
+  /** Service data */
+  service: services.Service;
+}
+
+/**
+ * Component that renders content for each service in the list.
+ * Maps over all services and provides each service through a service context.
+ * Only renders when services are successfully loaded (not loading, no error, and has services).
+ *
+ * @component
+ * @example
+ * ```tsx
+ * import { Services } from '@wix/services/components';
+ *
+ * function ServicesGrid() {
+ *   return (
+ *     <Services.ItemContent>
+ *       {({ service }) => (
+ *         <div className="service-card">
+ *           <img src={service.info.media?.mainMedia?.image?.url} alt={service.info.name} />
+ *           <h3>{service.info.name}</h3>
+ *           <p>{service.payment?.price} {service.payment?.currency}</p>
+ *           <button>Book Now</button>
+ *         </div>
+ *       )}
+ *     </Services.ItemContent>
+ *   );
+ * }
+ * ```
+ */
+export function ItemContent(props: ItemContentProps): React.ReactNode {
+  const {
+    services: servicesList,
+    isLoading,
+    error,
+  } = useService(ServicesListServiceDefinition);
+  const servicesValue = servicesList.get();
+
+  if (isLoading.get() || error.get() || servicesValue.length === 0) {
+    return null;
+  }
+
+  return servicesValue.map((service: services.Service) => (
+    <WixServices key={service._id} servicesMap={createServicesMap()}>
+      {typeof props.children === 'function'
+        ? props.children({ service })
```

Comment:
I wonder if we want to wrap with a "ServiceService" or ("BookingServiceService") for the entity itself. It could be that we'll need it later for entity mapping/formatting and more.